rust/chains/hyperlane-cosmos/src/providers/rpc.rs

```diff
@@ -84,84 +87,116 @@ impl WasmIndexer for CosmosWasmIndexer {
         Ok(result.block.header.height.value() as u32)
     }
 
-    async fn get_event_log<T>(
+    async fn get_range_event_logs<T>(
         &self,
-        block_number: u32,
+        range: RangeInclusive<u32>,
         parser: fn(Vec<EventAttribute>) -> Option<T>,
     ) -> ChainResult<Vec<(T, LogMeta)>>
     where
         T: Send + Sync,
     {
         let client = self.get_client()?;
-
-        let block = client.block(block_number).await?;
-        let block_result = client.block_results(block_number).await?;
-
-        let tx_hash: Vec<H256> = block
-            .block
-            .data
-            .into_iter()
-            .map(|tx| {
-                H256::from_slice(
-                    Hash::from_bytes(
-                        Algorithm::Sha256,
-                        hex::decode(digest(tx.as_slice())).unwrap().as_slice(),
-                    )
-                    .unwrap()
-                    .as_bytes(),
-                )
-            })
-            .collect();
-
-        let mut result: Vec<(T, LogMeta)> = vec![];
-        let tx_results = block_result.txs_results;
-
-        if tx_results.is_none() {
-            return Ok(result);
+        let contract_address = self.get_contract_addr()?;
+
+        let block_step: u32 = 100;
```

Comment:
I would prefer just using `range` and configuring the chunk to be 100 in the agent config - or whatever the max allowed results per page are. This allows us to rely on the logic that calls this function instead of iterating the range ourselves, because that logic already iterates over blocks and this way we also avoid querying non-existent blocks

Reply:
I think this may be because the max entries per page when getting blocks is 100 https://docs.tendermint.com/v0.34/rpc/#/Info/block_search so pagination is necessary

Fwiw, because there will be more blocks than messages, we maybe don't need to get the block hash of every block and just the ones with messages, but the # of RPCs is so low that it probably doesn't matter so much so I'm happy with the way it is for now